### DISCUSSION TITLE
The good way to get note absolute path  from root use ETAPI？

### DISCUSSION THREAD

**Discussion by holoword** [TriliumNext/Trilium#3774] [Q&A] [answered]
When I want to link a note to current note use ETAPI, Firstly I think /branches/{branchId}  get function in ETAPI, but I disappointed find it return nothing useful information.
eg: 
{'branchId': 'ttkIJGltmfMK_UlRVnmoHsWI3',
 'noteId': 'UlRVnmoHsWI3',
 'parentNoteId': 'ttkIJGltmfMK',
 'prefix': None,
 'notePosition': 10,
 'isExpanded': False,
 'utcDateModified': '2023-03-27 13:18:42.365Z'}
 the noteId and parentNoteId is all in branchId, prefix meaning has no description, blablabla
what I expect is to get the tree of a note, like '#root/ttkIJGltmfMK/UlRVnmoHsWI3', the note path from root.
Currently, I use /note/get method in ETAPI recursively find the path to root, Is there any other good way?

**Comment by zadam** [comment, marked as answer]
The complication of note cloning is that each branch can be part of several note paths.

An example of a branch:

* parentNoteId: Countries
* noteId: Austria

However, that parentNoteId can also have multiple branches with "Countries" as the child, e.g. "Europe" and "Holiday wishlist". So, there can be many different note paths leading to the same note or even same branch.

If you don't use cloning much, just use noteId as "note path" - trilium will resolve it to a full note path. Or you can use any "right suffix" - e.g. you can use parentNoteId/noteId and trilium will again resolve the rest.

Or is there a reason why you need a full note path?

**Reply by holoword** [reply]
I got the idea of /branches/{branchId} design for, but sounds don't like what I need.
Currently, I make a self use  PC send box which send note with file or link other note, while link other note or file, to process the link I use the full note path in the html.
eg, noteA fullpath is #root/UlRV/nmoH/sWI3, noteB fullpath is #root/ttkI/JGlt/mfMK. If I make a link to noteB in the noteA body, I use  #root/ttkI/JGlt/mfMK to link the note in the post request body. so as you say, I can use right suffix #JGlt/mfMK or other format other than fullpath?
I'm try the righ suffix, but sound like only fullpath work properly, is the format not right?
```html
<p><a href="#root/cV2Z2Q58MS5t/8Dg5C2c5fC6K/QX2oCY5lMDNG">Process</a></p>
<p><a href="#8Dg5C2c5fC6K/4efqbng8ZnoO">Process</a></p>
<p><a href="#8Dg5C2c5fC6K_4efqbng8ZnoO">Process</a></p>
<p><a href="8Dg5C2c5fC6K_4efqbng8ZnoO">Process</a></p>
```

**Reply by zadam** [reply]
So specifically in the URL you need to use also fixed `#root/` prefix and then the right suffix.

This requirement is specific to URLs to be able to easily recognize it's a note path.

So e.g. you can do these:

```html
<p><a href="#root/QX2oCY5lMDNG">Process</a></p>
<p><a href="#root/8Dg5C2c5fC6K/QX2oCY5lMDNG">Process</a></p>
```

**Reply by holoword** [reply]
Thanks a lot, I got the idea.
